ci/envs/311-dev.yaml

```diff
@@ -25,6 +24,7 @@ dependencies:
     # dev versions of packages
     - --pre --index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple --extra-index-url https://pypi.fury.io/arrow-nightlies/ --extra-index-url https://pypi.org/simple
```

Comment:
What if we include `--upgrade` here? That should enforce that we override whatever we have with nightlies, no?

Reply:
For the above reasons I don't think this will do anything, without upgrade we already are resolving for the latest numpy as nothing is installed, and then the requirement of contourpy means it is downgraded.